logdetective/server/database/models/merge_requests.py

```diff
@@ -0,0 +1,462 @@
+import enum
+import datetime
+from typing import Optional
+from sqlalchemy import (
+    Enum,
+    Column,
+    Integer,
+    DateTime,
+    String,
+    ForeignKey,
+    UniqueConstraint,
+    desc,
+)
+from sqlalchemy.orm import relationship
+from logdetective.server.database.base import Base, transaction
+
+
+class Forge(str, enum.Enum):
+    """List of forges managed by logdetective"""
+
+    gitlab_com = "https://gitlab.com"  # pylint: disable=(invalid-name)
+    gitlab_cee_redhat_com = "http://gitlab.cee.redhat.com/"  # pylint: disable=(invalid-name)
+
```

Comment:
On the one hand, I don't like having this as an `Enum`, because it inherently means that we need to modify the database schema in order to add another possible entry. An `Enum` is really intended to constrain values when there's a known, finite set that is valid. Whereas the set of possible Gitlab instances is potentially infinite, even if we only expect to handle a few.

On the other hand, it _does_ provide some protection from typographical errors (or someone trying to fill our DB table by sending us random `X-Gitlab-Server` headers to waste storage).

I suppose we can go with an Enum for now and fix things later if we grow to the point where it's a problem.

Reply:
Yes, but we should create a task about it, noting the tradeoffs.